include/seqan3/alphabet/composition/union_composition.hpp

```diff
@@ -461,39 +329,200 @@ class union_composition
+            for (auto k = 0u; k < value_sizes[i]; ++k, ++value)
+                value_to_char[value] = array_array[i][k];
+
+        return value_to_char;
+    }
+
+    /*!\brief Returns an map at compile time where the key is the char of one of the
+     * alphabets and the value is the corresponding rank over all alphabets (by
+     * conflict will default to the first).
+     *
+     * ```cpp
+     * constexpr auto char_to_value = char_to_value_table<char, dna4, gap, dna5>();
+     * assert(char_to_value.size() == 256);
+     * assert(char_to_value['A'] == 0);
+     * assert(char_to_value['C'] == 1);
+     * assert(char_to_value['G'] == 2);
+     * assert(char_to_value['T'] == 3);
+     * assert(char_to_value['-'] == 4);
+     * assert(char_to_value['A'] == 0);
+     * assert(char_to_value['C'] == 1);
+     * assert(char_to_value['G'] == 2);
+     * assert(char_to_value['T'] == 3);
+     * assert(char_to_value['N'] == 9);
+     * assert(char_to_value['*'] == 0); // every other character defaults to 0
+     * ```
+     */
+    static constexpr auto char_to_value_table() noexcept
+    {
+        constexpr auto value_size = alphabet_sum_size_v<alphabet_types...>;
+        using rank_t = detail::min_viable_uint_t<value_size>;
+
+        constexpr auto table_size = 1 << (sizeof(char_type) * 8);
+        constexpr auto value_to_char = value_to_char_table();
+
+        using array_t = std::array<rank_t, table_size>;
+
+        array_t char_to_value{};
+        for (auto i = 0u; i < value_to_char.size(); ++i)
+        {
+            auto & old_entry = char_to_value[value_to_char[i]];
+            auto is_new_entry = value_to_char[0] != value_to_char[i] && old_entry == 0;
+            if (is_new_entry)
+                old_entry = static_cast<rank_t>(i);
+        }
+        return char_to_value;
     }
 
     //!\brief Compile-time generated lookup table which contains the prefix sum up to the position of each alphabet
-    //!\hideinitializer
     //!\sa alphabet_prefix_sum_sizes
-    static constexpr auto prefix_sum_sizes =
-        detail::alphabet_prefix_sum_sizes<first_alphabet_type, alphabet_types...>();
+    static constexpr auto prefix_sum_sizes = alphabet_prefix_sum_sizes<alphabet_types...>();
 
     //!\brief Compile-time generated lookup table which maps the rank to char
-    //!\hideinitializer
     //!\sa value_to_char_table
-    static constexpr auto value_to_char =
-        detail::union_composition::value_to_char_table<char_type, first_alphabet_type, alphabet_types...>();
+    static constexpr auto value_to_char = value_to_char_table();
 
     //!\brief Compile-time generated lookup table which maps the char to rank
-    //!\hideinitializer
     //!\sa char_to_value_table
-    static constexpr auto char_to_value =
-        detail::union_composition::char_to_value_table<char_type, first_alphabet_type, alphabet_types...>();
+    static constexpr auto char_to_value = char_to_value_table();
```

Comment:
It would reduce the number of entities that you need to keep track of and and document...